### ISSUE TITLE
Separate the background of posts

### ISSUE THREAD

**Issue by eveneme** [WoltLab/WCF#6050]
I suggest separating each post with its own background instead of a combined one. In my opinion, it looks better and more readable because there aren't those harsh dividing lines between each post.

Before:
![image](https://github.com/user-attachments/assets/d881a6b6-d17c-4087-9b18-466c3b7d638a)

After:
![woltlab](https://github.com/user-attachments/assets/b33273d3-848c-4428-a9fb-2a91caffd308)

**Comment by eveneme** [comment]
@dtdesign @BurntimeX 

I would do something like this. In my opinion, there's currently too much use of borders, which could be replaced with shadows for a better visual effect. In my humble opinion, there are too many borders in the current design. The user sidebar could also be done in a similar way to the example in the attached screenshot. It's just a sample without most of the buttons and functions.

![Zrzut ekranu 2024-11-07 141021](https://github.com/user-attachments/assets/6c64dd38-f507-4797-b82a-1e268ec780b9)
![Zrzut ekranu 2024-11-07 141038](https://github.com/user-attachments/assets/b5ef65ef-f1e5-4713-b37c-2d9ffc5187ca)

**Comment by eveneme** [comment]
![image](https://github.com/user-attachments/assets/816b7ea6-aa6f-4aac-9f24-84f721223792)

**Comment by eveneme** [comment]
@BurntimeX @dtdesign 

Overall, I think a large user sidebar is unnecessary and a poor solution. The best option would be to create something in a similar style to what I’m showing below, which would allow the text to span the entire width. This is, of course, just an example, as it could be done better in terms of UI/UX. Please consider this.

![post](https://github.com/user-attachments/assets/4f4d07fe-2af4-4837-9b02-afb5b994241d)

**Comment by aze-dzn** [comment]
> @BurntimeX @dtdesign
> 
> Overall, I think a large user sidebar is unnecessary and a poor solution.

I don't like that version. You're losing a lot of space with that solution. How you want to display profile fields? If users are using more than one rank, there isn't enough space to display these, the same goes for trophies.

**Comment by eveneme** [comment]
@aze-dzn  I don’t agree with you; you're not losing space on the contrary, it actually looks better, and there’s room for ranks, etc. Just take a look. As for the current version, that's where you’re losing space. Look at how much empty space there is, especially with long messages. And in my example, you have the full width available for using trophy icons, post count, etc.

![image](https://github.com/user-attachments/assets/cd2cbac4-6df1-4714-bf3b-14886482e66c)

Plenty of unused space designated for nothing:

![Bez nazwy-1](https://github.com/user-attachments/assets/5ce8de61-04c6-4602-ab6d-306985cfdf1c)

Moving the addition time below this spot makes even more space, and in my opinion, it’s better designed in terms of UI/UX than a large empty field under the user sidebar.

![ersa](https://github.com/user-attachments/assets/a9d3f6f3-4ee4-4e55-94ce-4d3eb3b60bb2)

**Comment by cadeyrn** [comment]
> I suggest separating each post with its own background instead of a combined one.

I have also done this in my forums. I therefore agree with this proposal.

> Overall, I think a large user sidebar is unnecessary and a poor solution. The best option would be to create something in a similar style to what I’m showing below

I don't agree with that at all. The sidebar version is better in my opinion. It provides much more space for additional information. And sidebars are a good concept for _additional_ information which are not necessary to read. Also, you read from top to bottom and with your proposal, there is always information between the posts that is not relevant to the flow of reading. This is a classic case where function should take precedence over a supposedly modern look.

**Comment by eveneme** [comment]
@BurntimeX @dtdesign version v3:

![image](https://github.com/user-attachments/assets/dfb34207-0efe-45ea-abf9-3e9988cdac5a)

In my opinion, this layout is better designed in terms of UI/UX, with space being used more effectively compared to a side sidebar, where there is a lot of unnecessary space underneath. Please consider this. If there are any questions or suggestions, feel free to discuss.

> I don't agree with that at all. The sidebar version is better in my opinion. It provides much more space. Also, you read from top to bottom and with your proposal, there is always information between the posts that is not relevant to the flow of reading. This is a classic case where function should take precedence over a supposedly modern look.

Okay. And how does it look on a mobile device? :) The user avatar and so on are also on top. You read from left to right. 
The top bar takes up significantly less space than the left sidebar, which has a large empty area underneath, so the text spans the entire width of the box.

**Comment by cadeyrn** [comment]
> And how does it look on a mobile device? :)

On mobile, you have to deal with the different viewport size. It doesn't mean that you have to do the same on desktop. And as far as I can see, this is a proposal for the desktop design.

> You read from left to right.

I was not talking about the reading direction of sentences. You usually read posts from top to bottom. And on the desktop, with all the available space, there is no point in putting irrelevant information between these posts. The sidebar concept allows the user to easily skip the irrlevant information which is a good thing.

> The top bar takes up significantly less space than the left sidebar

I don't see the argument. Space is not an issue on the desktop at all. Again: The proposed top bar is a clear disadvantage for the reading flow and if you want to provide additional information, the sidebar provides more space for that.

**Comment by eveneme** [comment]
> > And how does it look on a mobile device? :)
> 
> On mobile, you have to deal with the different viewport size. It doesn't mean that you have to do the same on desktop. And as far as I can see, this is a proposal for the desktop design.
> 
> > You read from left to right.
> 
> I was not talking about the reading direction of sentences. You usually read posts from top to bottom. And on the desktop, with all the available space, there is no point in putting irrelevant information between these posts. The sidebar concept allows the user to easily skip the irrlevant information which is a good thing.
> 
> > The top bar takes up significantly less space than the left sidebar
> 
> I don't see the argument. Space is not an issue on the desktop at all. Again: The proposed top bar is a clear disadvantage for the reading flow and if you want to provide additional information, the sidebar provides more space for that.

Unfortunately, I don't agree with you on this matter. The user's sidebar takes up much more space than a small top bar. On Facebook, the user is also placed above each post, not on the side. Calculating in pixels, the narrow top bar takes up significantly less space than a side sidebar with a lot of unused space in the case of long messages.

![Bez nazwy-1](https://github.com/user-attachments/assets/41858ced-d044-44b4-9033-6b81d4d491d3)

**Comment by eveneme** [comment]
V4:
![image](https://github.com/user-attachments/assets/e3d29364-49fe-4d35-9eab-3013ba8d3c9b)
![image](https://github.com/user-attachments/assets/df4f1cc3-0894-411b-a796-70276507f2d4)

**Comment by dtdesign** [comment]
The top variant of your version 4 looks quite nice, although I would strongly prefer a different background color for the sidebar (maybe just a different shade). Our current mobile design uses a similar layout and the different background color serves as a visual guideline to quickly skim over a list of messages.

That said, I don’t see us abandoning the sidebar on the desktop view for two primary reasons:

1. There is a significant demand to display extra information alongside the message. The limited space with the sidebar at the top does not solve this, especially when dealing with text based information.
2. Removing the sidebar on the side on desktop causes the message to span most of the width of a screen.

Especially the display width is a major concern for us. As a rule of thumb, text shouldn’t be wider than 600-700 pixels because beyond that it becomes harder to read for many people, for example, it’s easier to lose track of the current line when wrapping. There are always people that have no issues reading super wide text rows but we need to design for the average user.

Moving the sidebar to the top is an option on desktop when either the layout width is heavily constrained and/or there is a permanent sidebar present that reduces the available text for the message. Pay close attention to the width of the content area on most platforms, it has a reason why they’re somewhat narrow.

**Comment by eveneme** [comment]
@dtdesign I understand; however, as for separating the background of posts, I believe it should be done this way. In my opinion, it looks better and more aesthetically pleasing than a solid, continuous background with <hr> dividers.

**Comment by dtdesign** [comment]
Yes, I agree. The current visuals are a result of a somewhat different approach in WoltLab Suite 3.0 which was subject to a significant change in WoltLab Suite 3.1. We’ve started adopting the card layout on more and more places as we overhauled and continue to overhaul parts of the software.

The message view is long outstanding for a complete visual and technical overhaul but somehow other (more urgent) things managed to get in the way. After all we have to take a lot more into consideration when deciding what to focus on next. The current view is looking quite dated but it’s still functionally getting the job done. Again, we can all agree that it is in need of an overhaul, I’m just explaining why we put our focus on other aspects first.

**Comment by eveneme** [comment]
@dtdesign @dtdesign  I would also clean up the appearance a bit; I would move the "edit" and "report content" buttons to a dropdown menu (three-dot icon) and leave only the essential buttons like reactions and quote. We could also add the text "quote," and it would be worth considering an "mention" button, which, when clicked, adds the tagged user's to the form text.

![extra](https://github.com/user-attachments/assets/5ecd63c6-aef1-487b-a8e3-3af1f944e2b1)

**Comment by eveneme** [comment]
@dtdesign Meanwhile, IPS V5 with the user displayed at the top, as I suggested earlier, looks more modern and compact in my opinion. Additionally, the sidebar with the user takes up too much space, and the avatar is also too large.

<img width="1467" height="606" alt="Image" src="https://github.com/user-attachments/assets/5b8b22de-e020-4924-b96b-f816bc4158d7" />

**Comment by BurntimeX** [comment]
Tbh, I don't like how IPS V5 does it. In a full-screen desktop window, the text area is too wide, which makes reading less comfortable. And opening the user statistics feels very clunky.

**Comment by dtdesign** [comment]
Yes, I have seen that layout and it does look quite tidy. That said, what I said above still stands, including my concerns with this kind of layout that I pointed out in https://github.com/WoltLab/WCF/issues/6050#issuecomment-2464300731 

The primary goal of a UI design is to provide a proper experience above all with visual appeal coming second. Neither does that mean it has to be neglected or that it’s okay to do a poor job there. The post view is the most important part in our software and regardless of the direction in which we will be moving, design must follow function at all cost here.

Interestingly, blogs and news site have understood this key principle for a long time already but in the world of forums having a line of text running from the left side of the screen to the right one is still considered acceptable. No, it is not. This isn’t even about individual taste, it’s entirely about readability and super long lines are always bad.

**Comment by eveneme** [comment]
@dtdesign In IPS V5 you can choose between the post layout I showed above or the classic one with a user sidebar. Maybe the key is to create a template with more customization options? That could be the way to satisfy both sides.

**Comment by dtdesign** [comment]
That is an obvious path we have considered too but it does not solve the problem.

1. Choice comes at a price. Offering two variants means that there are two layout options that have to be maintained by both us and style creators. Also if you cannot lock down the value in one way or another this creates problems when the layout really is only designed with the sidebar at location X but the user changes this to location Y. Locking in the option means that users no longer have the choice if they prefer the other variant.
2. Putting the sidebar at the top and allowing the text to take up the full horizontal space without restricting it is a bad idea. This is an issue that I have pointed out above already and is something we’re not willing to ignore. We have to make design choices with the actual users in mind because the customer is only right in matters of taste and this is about usability, not taste! ;)

**Comment by eveneme** [comment]
@BurntimeX Is there also a plan to redesign this section, or is it only about separating the background?
